esp/bindings/http/platform/httpservice.cpp

```diff
@@ -1176,17 +1176,22 @@ bool CEspHttpServer::isAuthRequiredForBinding(EspAuthRequest& authReq)
     return true;
 }
 
-EspAuthState CEspHttpServer::checkUserAuthPerSession(EspAuthRequest& authReq)
+EspAuthState CEspHttpServer::checkUserAuthPerSession(EspAuthRequest& authReq, StringBuffer& authorizationHeader)
 {
     ESPLOG(LogMax, "checkUserAuthPerSession");
 
     unsigned sessionID = readCookie(authReq.authBinding->querySessionIDCookieName());
     if (sessionID > 0)
         return authExistingSession(authReq, sessionID);//Check session based authentication using this session ID.
 
-    if ((authReq.authBinding->getDomainAuthType() != AuthPerRequestOnly) && authReq.authBinding->isDomainAuthResources(authReq.httpPath.str()))
+    if (authReq.authBinding->isDomainAuthResources(authReq.httpPath.str()))
         return authSucceeded;//Give the permission to send out some pages used for login or logout.
 
+    if (!authorizationHeader.isEmpty() && (authReq.serviceName.isEmpty() || authReq.methodName.isEmpty()
+        || !strieq(authReq.serviceName.str(), "esp") || (!strieq(authReq.methodName.str(), "login")
+        && !strieq(authReq.methodName.str(), "unlock"))))
```

Comment:
A little hard to follow that your really just checking if it’s one of those two methods, but ok.

Reply:
Agree with Tony. Please refactor or add comments explaining what you are doing here